lib/javascript/JavascriptParser.js

```diff
@@ -4128,6 +4128,22 @@ class JavascriptParser extends Parser {
 	 * @returns {void}
 	 */
 	inBlockScope(fn, inExecutedPath = false) {
+		let enterTry = false;
+
+		loop: for (let i = this.statementPath.length - 1; i >= 0; i--) {
+			const statementPath = this.statementPath[i];
+			switch (statementPath.type) {
+				case "TryStatement":
+					enterTry = true;
+					break loop;
+				case "BlockStatement":
+					continue;
+				default:
+					// we meet other statement, we are not directly in try block
+					break loop;
+			}
+		}
```

Comment:
I am afraid it can be slow, can we do it only when necessary?

Reply:
Especially in very big and nested code

Reply:
Or maybe we can refactor our logic and break only in statements?

Reply:
This should not be slow as it will break immediately when meets any statement that is not `BlockStatement`, it can only be slow if user code like `;{{{{ statement; }}}}`